include/cctz/civil_time_detail.h

```diff
@@ -367,11 +367,11 @@ class civil_time {
       : civil_time(ct.f_) {}
 
   // Factories for the maximum/minimum representable civil_time.
-  static CONSTEXPR_F civil_time (max)() {
+  static CONSTEXPR_F auto (max)()->civil_time {
```

Comment:
> The compiler cannot tell that (max) is a member function, NOT a macro. A trailing return type would fix this.

I'm not sure how a trailing return type changes the interpretation of the `(max)` tokens and the application of the macro, but that's OK.  If you have more detailed insight, I'd be happy to hear it.

But let's at least change this to surround the `->` with spaces, here and below, so as to match other usages within the library.  Thanks.

Reply:
Thank you for the comment.  I am not sure either, but it did fix it for MSVC 17.7.1 and above (for earlier versions MSVC 17.6.x the original code without change worked no problem).  MSVC 17.7 introduced new features on MACRO treatment, maybe those broke things.

Reply:
OK, thanks.